bulk_handlers_test.go

```diff
@@ -40,3 +41,185 @@ func TestBulkCreateMissingSourceType(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+func TestCreateUserWithResourceOwnershipApplicationType(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
+
+	conf.ResourceOwnership = "user"
+
+	testUserId := "testUser"
+	identityHeader := testutils.IdentityHeaderForUser(testUserId)
+
+	nameSource := "test source"
+	sourceTypeName := "bitbucket"
+	applicationTypeName := "app-studio"
+	authenticationResourceType := "application"
+
+	requestBody := testutils.SingleResourceBulkCreateRequest(nameSource, sourceTypeName, applicationTypeName, authenticationResourceType)
+
+	body, err := json.Marshal(requestBody)
+	if err != nil {
+		t.Error("Could not marshal JSON")
+	}
+
+	c, _ := request.CreateTestContext(
+		http.MethodPost,
+		"/api/sources/v3.1/bulk_create",
+		bytes.NewReader(body),
+		map[string]interface{}{
+			"tenantID": int64(1),
+		},
+	)
+
+	c.Request().Header.Add("Content-Type", "application/json;charset=utf-8")
+	c.Set("identity", &identity.XRHID{Identity: identityHeader})
+
+	err = BulkCreate(c)
+	if err != nil {
+		t.Error(err)
+	}
+
+	var users []m.User
+	err = dao.DB.Model(&m.User{}).Find(&users).Error
```

Comment:
I think here we can have problem in the future when we add fixtures for users into tests or if different test creates new users ... now the test relies on the fact that in "users" table is only 1 record (and this one record is "our" user)